src/NosCore.Controllers/GroupPacketController.cs

```diff
@@ -0,0 +1,330 @@
+                    {
+                        Session.SendPacket(new MsgPacket
+                        {
+                            Message = Language.Instance.GetMessageFromKey(LanguageKey.GROUP_BLOCKED, Session.Account.Language)
+                        });
+                        return;
+                    }
+
+                    Session.Character.GroupRequestCharacterIds.Add(pjoinPacket.CharacterId);
+
+                    if (Session.Character.Group == null || Session.Character.Group?.Type == GroupType.Group)
+                    {
+                        if (targetSession.Character.Group == null || targetSession.Character?.Group?.Type == GroupType.Group)
+                        {
+                            Session.SendPacket(new InfoPacket { Message = Language.Instance.GetMessageFromKey(LanguageKey.GROUP_INVITE, Session.Account.Language) });
+                            targetSession.SendPacket(new DlgPacket
+                            {
+                                Question = Language.Instance.GetMessageFromKey(LanguageKey.INVITED_YOU_GROUP, targetSession.Account.Language),
+                                YesPacket = new PjoinPacket { CharacterId = Session.Character.CharacterId, RequestType = GroupRequestType.Accepted },
+                                NoPacket = new PjoinPacket { CharacterId = Session.Character.CharacterId, RequestType = GroupRequestType.Declined }
+                            });
+                        }
+                    }
+
+                    break;
+                case GroupRequestType.Sharing:
+
+                    if (Session.Character.Group == null)
+                    {
+                        return;
+                    }
+
+                    Session.SendPacket(new InfoPacket
+                    {
+                        Message = Language.Instance.GetMessageFromKey(LanguageKey.GROUP_SHARE_INFO, Session.Account.Language)
+                    });
+
+                    Session.Character.Group.Characters.Values.Where(s => s.Character.CharacterId != Session.Character.CharacterId).ToList().ForEach(s =>
+                    {
+                        s.Character.GroupRequestCharacterIds.Add(s.Character.CharacterId);
+                        s.SendPacket(new DlgPacket
+                        {
+                            Question = Language.Instance.GetMessageFromKey(LanguageKey.INVITED_GROUP_SHARE, Session.Account.Language),
+                            YesPacket = new PjoinPacket { CharacterId = Session.Character.CharacterId, RequestType = GroupRequestType.AcceptedShare },
+                            NoPacket = new PjoinPacket { CharacterId = Session.Character.CharacterId, RequestType = GroupRequestType.DeclinedShare }
+                        });
+                    });
+
+                    
+                    break;
+                case GroupRequestType.Accepted:
+                    if (targetSession == null)
+                    {
```

Comment:
if this is true we should check it only once at the begining and not enter in this